### PR TITLE
Start / Finish Course Subject by Supervisor

### PR DESCRIPTION
**Pull request by abdullahallmehedi**

### CODE REVIEW

**Comment by Rownaknira** [comment]
ok

File: app/controllers/supervisor/courses_subjects_controller.rb [hunk 1]
```diff
@@ -1,7 +1,11 @@
 class Supervisor::CoursesSubjectsController < ApplicationController
   before_action :load_courses_subject
+<<<<<<< HEAD
```
Comment (dieunb) [inline app/controllers/supervisor/courses_subjects_controller.rb]: remove it

File: app/controllers/supervisor/courses_subjects_controller.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
   before_action :check_courses_subject
   load_and_authorize_resource :course
+=======
```
Comment (dieunb) [inline app/controllers/supervisor/courses_subjects_controller.rb]: remove it

File: app/controllers/supervisor/courses_subjects_controller.rb [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
+  before_action :check_courses_subject, only: [:edit, :update]
+>>>>>>> Start / Finish Course Subject by Supervisor
```
Comment (dieunb) [inline app/controllers/supervisor/courses_subjects_controller.rb]: remove it

File: app/controllers/supervisor/courses_subjects_controller.rb [hunk 4]
```diff
@@ -27,13 +37,32 @@ def load_courses_subject
...
+    if @courses_subject.new?
+      @courses_subject.status = CoursesSubject::STATUS[:started]
+    elsif @courses_subject.started?
+      @courses_subject.status = CoursesSubject::STATUS[:finished]
+    end
+    @courses_subject.save!
   end
 
   def check_courses_subject
+<<<<<<< HEAD
```
Comment (dieunb) [inline app/controllers/supervisor/courses_subjects_controller.rb]: remove it

**Pull request merged**